best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,285 @@
+
+### EST Server - Client Authentication
+
+The EST Server MUST authenticate the EST Client that is requesting a TLS Certificate manually or automatically using a TLS Client Certificate.
+
+The EST Server MUST support using a TLS Client Certificate, presented during the TLS handshake by the EST Client to authenticate if the Client is trusted. The TLS Client Certificate can either be signed by the current CA or an externally trusted Root CA. The EST Server MUST check the validity of the EST CLients TLS Certificate before responding to its request. The EST Server MUST provide a method to load multiple trusted Root CA's, that are used to verify TLS Client Certificate.
+
+The EST Server MAY also support manually authentication of the EST Client if; no TLS Client Certificate is presented during the TLS handshake, the TLS Client Certificate is not trusted or as an extra authentication step. The exact process for manual authentication will be implementation specific, but the EST Server MUST provide enough information to the operator so they can authenticate the EST Client. During the manual authentication the EST Server MUST respond with either HTTP 202 or HTTP 503 and the response must include a `Retry-After` header.
+
+The EST Server MUST be capable of issuing TLS Certificates signed with both RSA and ECDSA keys.
+
+The EST Server MUST return a TLS Certificate with the Extended Key Usage set for both TLS Server Authentication and TLS Client AUthentication as per [RFC 5280][RFC-5280]. This is to allow the TLS Certificate to be used to authenticated the NMOS APIs to NMOS Clients and for authentication of the EST Client during certificate renewal.
+
+## EST Client
+
+An EST Client SHOULD allow EST to be disabled, preventing the EST Client from being automatically provisioned with a TLS Certificate if required by the networks security policy.
+
+An EST Client SHOULD allow manual configuration of the EST Servers Hostname and Port, to prevent the EST Client from requesting a TLS Certificate from a rogue server.
+
+An EST Client MUST provide a method to manually install both the Root CA and TLS Server certificate for the target environment, for the case when an EST Server is not present or the TLS Client Certificate is no longer valid.
+
+If the EST Server returns a HTTP re-direct, the the EST Client SHOULD follow the re-direct URL.
+
+If the EST Server returns an HTTP 4xx, HTTP 5xx or Connection Timeout, the EST Client SHOULD attempt the request again using an alternative EST Server if present, else the EST Client SHOULD wait an appropriate exponential backoff period before retrying.
+
+### Initial Certificate Provisioning
+
+**EST Server Discovery**
+
+On connection to the target environments network the EST Client SHOULD attempt to discover the location of the EST Server using [DNS-SD](#dns-sd-advertisement), if EST has not been disabled or the location of the EST configured manually.
+
+**Get Root CA**
```

Comment:
We should probably include explicit guidance about refreshing this too.

Reply:
Whilst the default for refreshing this may be as it approaches expiry, I think we also need a section regarding revocation. If the EST Client is also an NMOS Client it may identify that the CA cert has been revoked at some point, which should be an indication it needs to go through initial enrolment again presumably.

Similarly, if the server cert is revoked, the EST Client won't necessarily be notified that this has happened. We may need to mandate some mechanism to force the certificate renewal in this situation.

Reply:
Added a section certificate revocation